Feature/searchfunction

DErdmann1 opened the pull request:
Contrary to my user story, I did not add a magnifying glass icon, but instead directly display the input field. Perhaps this will be changed in a later user story that deals with styling. If no results are found, a message is displayed. Upper- or lowercase is irrelevant in the function.

vaupunkt commented on components/Doglist/index.js:
```diff
@@ -1,9 +1,19 @@
-import Image from "next/image";
+import React from "react";
```
you don't need to import react :)

vaupunkt commented on components/Catlist/index.js:
```diff
@@ -1,8 +1,15 @@
-import Image from "next/image";
+import React from "react";
```
you don't need to import react :)

vaupunkt reviewed (changes requested):
i think you don't need three list-components and three search-functions.
in the end you can simplify it! 

in your cats, dogs or smallanimals-page you already have the filtered list for the animals. you could take this filtered list to one list-component, to render the filtered animals and retake the same component for the other animal lists. the same with the search. because you already work with filtered lists the search would work the same for all three animal-lists 😊

DErdmann1 commented:
I will combine the animal lists and the filtering function in another user story. However, I have deleted my React imports and adjusted the search function to update the results after each new input. :)

AliaksandraNovikava commented on components/Bookmarkbutton/index.js:
```diff
@@ -1,29 +1,25 @@
       {isBookmarked ? (
         <svg
           xmlns="http://www.w3.org/2000/svg"
           viewBox="0 0 24 24"
           width="24"
           height="24"
         >
-          <title>bookmark</title>
-          <path d="M17,3H7A2,2 0 0,0 5,5V21L12,18L19,21V5A2,2 0 0,0 17,3Z" />
+          <title>star</title>
+          <path d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z" />
         </svg>
```
You could consider creating an external svg file and save your svg code there and then import it as component. You could reduce your code significantly🤓

AliaksandraNovikava reviewed (approved):
Hey @DErdmann1 
I only checked your new changes.
Like Veliko mentioned, I would also recommend to reduce your three Searchfunction components, that are practically the same, to one. But you already said that you are planning to do so👍
I left you one comment regarding svg implementation🤓
